### ISSUE TITLE
Wav2Vec 2.0 pretraining phase speed

### ISSUE THREAD

**Issue by jubick1337** [pytorch/fairseq#3594]
#### What is your question?
I'm trying to find the fastest way to do w2v2 pretraining (complete 400k updates). 
I've tried several machines with different configurations.
I feel like using more GPUs would give speed boost (num_updates/time) but results are opposite.
Correct me if I'm doing something wrong.

#### Code
command to run train:
fairseq-hydra-train task.data=/root/micro/ --config-dir fairseq/examples/wav2vec/config/pretraining/ --config-name wav2vec2_base_librispeech.yaml
Inside of config I've changed num_workers and distributed world size to match machine params.

#### What have you tried?
Experimented!
Standard way: 4 minutes 8 seconds to 5 epochs and 110 train updates, train loss 9.57
As above but distributed world size==1 2 minutes 54 seconds to 5 epochs and 440 train updates, train loss 6.807
Standard way but optimization.update_freq='[16]'  3 minutes 47 seconds to 5 epochs and 9 train updates, train loss 11.644

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): fairseq repo at 3084b812beb72a880b6ddb5d9076ead60b7232d6
 - PyTorch Version (e.g., 1.0) 1.9.0a0+2ecb2c7
 - OS (e.g., Linux): Ubuntu 20.04 LTS
 - How you installed fairseq (`pip`, source): source
 - Build command you used (if compiling from source): 
 - Python version: 3.8
 - CUDA/cuDNN version: 8.2.0.51
 - GPU models and configuration: 4x RTX 3090
 - Any other relevant information: I'm using this [image](https://ngc.nvidia.com/catalog/containers/nvidia:pytorch) as it has preinstalled apex and other Nvidia stuff. I've tried stable version of Pytorch (1.8.0) and no apex but they were much slower.
